metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -879,6 +879,79 @@ def _extract_lineages(self):
         return
 
 
+@dataclass
+class SnowflakeSinkConnector:
+    connector_manifest: ConnectorManifest
+    report: KafkaConnectSourceReport
+
+    def __init__(
+        self, connector_manifest: ConnectorManifest, report: KafkaConnectSourceReport
+    ) -> None:
+        self.connector_manifest = connector_manifest
+        self.report = report
+        self._extract_lineages()
+
+    @dataclass
+    class SnowflakeParser:
+        database_name: str
+        schema_name: str
+        topics_to_tables: Dict[str, str]
+
+    def report_warning(self, key: str, reason: str) -> None:
+        logger.warning(f"{key}: {reason}")
+        self.report.report_warning(key, reason)
+
+    def get_parser(
+        self,
+        connector_manifest: ConnectorManifest,
+    ) -> SnowflakeParser:
+        database_name = connector_manifest.config["snowflake.database.name"]
+        schema_name = connector_manifest.config["snowflake.schema.name"]
+        topics_to_tables: Dict[str, str] = {}
+
+        # If config contains topic to table map, take target table from config
+        # Else kafka connect create target table with same name as source topic
+        if "snowflake.topic2table.map" in connector_manifest.config:
+            for each in connector_manifest.config["snowflake.topic2table.map"].split(
+                ","
+            ):
+                topic, table = each.split(":")
+                # Extract lineage for only those topics whose data ingestion started
+                if topic in connector_manifest.topic_names:
+                    topics_to_tables[topic] = table
```

Comment:
In case whitespace is allowed between commas or colons.
```suggestion
                if each.strip():
                    topic, table = each.split(":")
                    # Extract lineage for only those topics whose data ingestion started
                    if topic in connector_manifest.topic_names:
                        topics_to_tables[topic.strip()] = table.strip()
```